### PR TITLE
test: add integration test for declining an invite to an empty room

### LINKED ISSUE #4642: Impossible to decline an invite to an empty room
- Create a private room
- invite someone
- make them admin too (not sure if this is necessary)
- leave the room before the other person can accept the invite

Result:
- The other user now has an invite in the room list, but it can't be accepted or declined and never goes away `403 FORBIDDEN`

### PR DESCRIPTION
**Pull request by paul8711-code**
This PR is related to #4642. I have added the first of two integration tests (as requested), according to which declining an invite worked just fine.

- [ ] I've documented the public API changes in the appropriate changelog files (see [Writing changelog entries][pull-request-template-changelog]).
- [ ] This PR was made with the help of AI.

[pull-request-template-changelog]: https://github.com/matrix-org/matrix-rust-sdk/blob/main/CONTRIBUTING.md#writing-changelog-entries

Signed-off-by: Paul8711 <154304377+paul8711-code@users.noreply.github.com>

### CODE REVIEW

**Comment by vektrace** [comment]
@Hywan could you take a look at this PR when you have time? It is directly relevant to our discussion in the previously mentioned issue.

File: testing/matrix-sdk-integration-testing/src/tests/room.rs [hunk 1]
```diff
@@ -36,6 +36,52 @@ use tracing::{debug, error, warn};
...
+        }))
+        .await?
+        .room_id()
+        .to_owned();
+
+    if let Some(room) = alice.get_room(&room_id) {
+        room.leave().await?;
+    }
+
+    if let Some(room) = bob.get_room(&room_id) {
```
Comment (Hywan) [inline testing/matrix-sdk-integration-testing/src/tests/room.rs]: Bob probably hasn't sync yet, or didn't receive the invite, so `get_room` will return `None` and the test will pass.

Please ensure the room exists to get a real test!

**Review by Hywan — COMMENTED** [review]
Thanks for the test! I think we need to tweak it though, as it contains a flow right now.

File: testing/matrix-sdk-integration-testing/src/tests/room.rs (hunk 1 above)
Comment (vektrace) [reply testing/matrix-sdk-integration-testing/src/tests/room.rs]: This should have been resolved now, I used the approach from the test below mine to make sure Bob received the invite.

File: testing/matrix-sdk-integration-testing/src/tests/room.rs [hunk 2]
```diff
@@ -36,6 +36,61 @@ use tracing::{debug, error, warn};
...
+    let room_id = alice
+        .create_room(assign!(CreateRoomRequest::new(), {
+            invite: vec![bob.user_id().unwrap().to_owned()],
+            is_direct: false,
+        }))
+        .await?
+        .room_id()
+        .to_owned();
+
+    if let Some(room) = alice.get_room(&room_id) {
```
Comment (Hywan) [inline testing/matrix-sdk-integration-testing/src/tests/room.rs]: I guess we should adopt a similar pattern here:

```rust
let room = alice.get_room(&room_id).expect("Alice should see the room");
room.leave().await?;
```

**Review by Hywan — CHANGES_REQUESTED** [review]
Almost good!

**Review by Hywan — APPROVED** [review]
Looks good to me, thanks!
